diff --git a/arch/x86/kvm/mmu/mmu.c b/arch/x86/kvm/mmu/mmu.c
index 5e25d93ec..e42914034 100644
--- a/arch/x86/kvm/mmu/mmu.c
+++ b/arch/x86/kvm/mmu/mmu.c
@@ -3551,8 +3551,8 @@ static bool get_mmio_spte(struct kvm_vcpu *vcpu, u64 addr, u64 *sptep)
 		 * reserved bit and EPT's invalid memtype/XWR checks to avoid
 		 * adding a Jcc in the loop.
 		 */
-		reserved |= __is_bad_mt_xwr(rsvd_check, sptes[level - 1]) |
-			    __is_rsvd_bits_set(rsvd_check, sptes[level - 1],
+		reserved |= (int)__is_bad_mt_xwr(rsvd_check, sptes[level - 1]) |
+			    (int)__is_rsvd_bits_set(rsvd_check, sptes[level - 1],
 					       level);
 	}
 
diff --git a/tools/perf/bench/inject-buildid.c b/tools/perf/bench/inject-buildid.c
index 280227e3f..fbcb9d8da 100644
--- a/tools/perf/bench/inject-buildid.c
+++ b/tools/perf/bench/inject-buildid.c
@@ -378,6 +378,7 @@ static int inject_build_id(struct bench_data *data, u64 *max_rss)
 	wait4(data->pid, &status, 0, &rusage);
 	*max_rss = rusage.ru_maxrss;
 
+	(void)len;
 	pr_debug("   Child %d exited with %d\n", data->pid, status);
 
 	return 0;
